### ISSUE TITLE
Hide more features on Kiosk mode

### ISSUE THREAD

**Issue by raelgc** [RetroPie/EmulationStation#431]
Hi guys,

I've configured some boxes to work in the Kiosk mode, but still kids are messing with them. Then, when the next kid will start to play, a lot of the configurations are changed.

Basically my idea is to hide more options/features in Kiosk mode:

1 - hide the entire main menu (the one fired with <kbd>start</kbd>): some kids will put sound volume to `0` or will shutdown the system;

2 - on game list view, disable `Favorites` (to avoid add/remove items to favorites) and `Options` (keep the game list display settings untouched);

3 - on system list view, hide `Launch Screensaver` entry. My ScreenSaver is the black one. Some kids start it and then came to ask why "nothing is displayed".

I can work on this if you all approve the idea.

**Comment by pjft** [comment]
I'd like to get @zigurana 's perspective on this, but in summary the idea would likely be that if it's a new mode it will likely have a better chance of getting approved.

There's a reason why volume is there, as on handhelds or devices without volume control that's the only way to actually change it.

For the launch screensaver entry I'm fairly sure you can disable that setting in the menu (screensaver controls, I think?). See if that helps.

The favorites I think it's a good point. In fact, I wonder if it should be an option for all modes, actually - when it was designed I imagined it would be more frequently used but in truth I find that after the favorites are set it's not as common to go there and change them. I know @zigurana left that there on purpose, so I'm thinking of changing that on the collections options instead.

Thoughts? I suppose it's only the volume that's left. I wouldn't change the current kiosk mode but unsure of the best approach.

**Comment by raelgc** [comment]
Well, imho a kiosk mode should really act as a kiosk and be "untouchable". Actually it's more close to a "Simple Options" mode.

Anyway, let me think in line with your thoughts for a possible second approach.

Screensaver: just found the option to hide controls now. For some reason, the **ScreenSaver Controls** option lives under **Main Menu** > **UI Settings** > **ScreenSaver Settings** > **Video ScreenSaver Settings**. It should be one level above, not under Video. Anyway, it works. What I can do here is move it a level up.

For sound, I can realize the importance for devices without volume control (but again, a mode called kiosk should really be untouchable). But I can live with it :)

Favorites and GameList options controls: they really should be not displayed in kiosk mode.

Quit: We already have a "--no-exit" parameter, but for some reason it hides only the "Exit EmulationStation" option. I think it should hide the entire "Quit" submenu. A kiosk mode should not allow shutdown or restart.

**Comment by pjft** [comment]
Do move it one level up, must've been lost in a refactoring. Thanks.

In regards to exiting, though, how would you actually shut down the device? I understand the use case, though the use case implemented here is a middle ground, for home use. It's not running 24/7, and in specific for the Pi there's no real way to shut it down (unless you own a power button GPIO thingie).

I personally don't oppose removing favorites from Kiosk mode, but once again, it was not my design so I'd rather have @zigurana commenting on that change and whether the use cases would be broken or not. Truth is that this has been around for a while now, and while I understand the validity of your use cases, the truth is that the current design may support use cases that people have been using so far that your changes will break. Hence my reluctance in blanket approving it.

The favorites thing is certainly something I can comment on and chime in with more confidence.

**Comment by dankcushions** [comment]
personally i would remove shut down from kiosk mode, also. i keep my pi on 24/7 at home (doesn't everyone?!) and even if i didn't, i view 'kiosk' as 'unattended' mode, like if you had a bartop and left it running with friends over. i would want to be able to leave it knowing that the only thing it could be used for is playing games, and no settings could be changed. i would remove volume, also. i'd remove the whole menu!

**Comment by raelgc** [comment]
@dankcushions exactly my thought. Maybe like @pjft commented, a new mode, like `arcade` or `unattended`.

**Comment by pjft** [comment]
I won't object to any of these strongly, other than asking for really appreciating @zigurana 's perspective.

Truth be told, volume is the main one I'd miss there in that case, but I admit it might not be a strong enough case for it to be kept. I do believe that Volume was explicitly asked by someone, as it was not part of the original design.

Any objections on it being a separate mode - or, alternatively, turning "kiosk" into these settings and creating a differently-titled mode with the previous settings?

I can take both for an answer, ultimately. Just probing for thoughts.

Thanks both.

**Comment by raelgc** [comment]
@pjft Probably the current kiosk should be something like "Simple". So we'll have Kiosk (a real unnatended mode), Simple (the current kiosk) and Full.

Just need to figure a way to not break people upgrading. I.e., people already using the current Kiosk that should be, after upgrade, get into the Simple.

About the other approach for current Kiosk: we can keep volume, hide Favorites and ListOptions and introduce a `--no-quit` to disable the Quit submenu.

**Comment by pjft** [comment]
I don't object to the first option. If the concern is about upgrading, we can instead name the new mode "Full Kiosk" or "Locked Kiosk" or something.

Thanks.

**Comment by zigurana** [comment]
Hey guys, (thanks @pjft for tagging me! )
What to include/exclude in the different UI modes (Full/Kiosk /Kid) came up during the initial development, and along much the same lines. 
My guiding principle for Kiosk mode has always been that users in Kiosk mode should not be able to damage the system, while still be able to access all games. Now what you all consider 'damage' is up for discussion. I consider sound volume levels and favorites to be harmless settings, but you are of course free to disagree.
The shutdown option was retained for much the same reason, as even after rebooting the system, no damage would have occurred (other than a loss of time).

Those were my considerations, and I will certainty understand if other people have different use cases. 

In the end, the real solution for this is create an UI where the visibility of each element, menu, shortcut is fully configurable.
This of course is out of scope for the current discussion, but there will be others with different uses, and different requirements, and we will have another repetition of this discussion. :-)

**Comment by raelgc** [comment]
Well, how about name this unattended mode `arcade`, then? In an arcade you cannot change volume, shutdown, add/remove favorites, etc.

**Comment by zigurana** [comment]
Sure, I mean you dont need my permission to make anything you want :-). 
I just want to warn against creating a never-ending list of UI modes with varying degrees of exposure of the full UI.

Now, if you think there is a consensus on this, and that the name 'arcade' is descriptive enough to make it's use clear to users, go for it!

**Comment by pjft** [comment]
@zigurana great to hear from you, hope you're doing well!

If you don't have any objections, as you were the one creating this initially, I'm fine either way - be it a new mode or changing the current one. Just wanted to make sure that your original use case wasn't broken.

Best!

**Comment by HerbFargus** [comment]
One thing retroarch has done is made the menu configurable so that users can disable any portion of the menu they like (though they require a manual modification of a config file to bring them back), just another idea. or could just have a tier of all settings, no settings, and then somewhere in between - I believe this is pretty much what we discussed the last time we went through this

**Comment by raelgc** [comment]
@zigurana What do you think if instead of introduce a new mode I do the following in current kiosk mode:

1. hide/disable favorite controls and list-view controls
2. add a `--no-quit` parameter, similar to `--no-exit`
3. move the ScreenSaver Control show/hide button one level up (it's under Video ScreenSaver Settings)

**Comment by zigurana** [comment]
Hey Herb! That route is indeed the only real way around this issue once and for all. It's a mayor effort though, so a long term goal, at best.

@raelgc: I agree that introducing is maybe a bit overkill, especially if it is not entirely clear what a certain UI exposes or not.
Looking again at your first post, it seems that your users are mostly children that have some difficulty understanding /navigating the UI. What do you think of implementing your changes in Kids mode? That way, there is a more or less natural progression in the amount of shielding that happens from Full to Kiosk to Kids.
While hiding the main and options menus completely will not fly in the kiosk mode, in Kid I would find it perfectly acceptable.

**Comment by raelgc** [comment]
I'm fine implementing this in kids mode, @zigurana. Nice suggestion, indeed!

The only effort for someone creating a bartop/arcade will be tagging all games as a kidgame (really simple to achieve using command line or even a find/replace tool in any text editor).

**Comment by zigurana** [comment]
Yeah, that's a pain, for which a command - line switch could be introduced (--force_show_all or somesuch), if you want to save yourself and others from editing their metadata. Good luck, i will be happy to have a look at your PR in due time.

**Comment by raelgc** [comment]
For the above PR, aside to the changes to kids mode (hide main menu, screensaver), I've added a `--force-skip-filters` option. So, for an arcade mode, we can use `emulationstation --force-kid --skip-filters`. 

Additionally I've moved the ScreenSaver Controls toggle one level up.

**Comment by pjft** [comment]
Thanks.

Just a note, probably personal preference but I think it's good as this is ultimately the RetroPie repository: I'm usually not be thrilled with adding a command-line option without a paired menu option for it. The majority of the users of this ES fork will be people who install it and use it in RetroPie. Given that they don't have an easy way to use those options, I'd support implementing the corresponding option under "Other Options" or somewhere where it'd fit.

I know that there are some options that are command-line only (some coming from the original Aloshi's version, and others that might have come up later) but consider that.

Thanks for your hard work here!

**Comment by raelgc** [comment]
@pjft Following your request, I've added a `ENABLE FILTERS` entry under **UI Settings**. I've renamed the internal setting entry to be `ForceDisableFilters` (as parameter, `--force-disable-filters`).

When this option is changed throw the UI, I'm using `ViewController::get()->ReloadAndGoToStart()` as this will remove filters for all systems. Additionally when filters are disabled, I'm hidding the **Filter Gamelist** entry in the *Options* UI to be consistent.
